0.6.0 SIGSEGV on Control-C shutdown

#### Summary
Just fired up my first 0.6.0 node on Ubuntu running without bee-clef and starting interactively with "bee start".  When I hit Control-C to shut the node down, it terminates with SIGSEGV:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xf82e57]

goroutine 23148 [running]:
github.com/ethersphere/bee/pkg/p2p/libp2p.(*Service).Close(0xc0004aa9c0, 0x0, 0x0)
        github.com/ethersphere/bee/pkg/p2p/libp2p/libp2p.go:706 +0x57
github.com/ethersphere/bee/pkg/node.(*Bee).Shutdown(0xc000492240, 0x1705968, 0xc002e064e0, 0x1705968, 0xc002e064e0)
        github.com/ethersphere/bee/pkg/node/node.go:693 +0x49d
github.com/ethersphere/bee/cmd/bee/cmd.(*command).initStartCmd.func1.2.1(0xc00015e780, 0xc000492240, 0x1715f20, 0xc000166600)
        github.com/ethersphere/bee/cmd/bee/cmd/start.go:180 +0xaa
created by github.com/ethersphere/bee/cmd/bee/cmd.(*command).initStartCmd.func1.2
        github.com/ethersphere/bee/cmd/bee/cmd/start.go:174 +0xac


#### Steps to reproduce
Run the node with "bee start" and hit Control-C after it is running for a while.

#### Expected behavior
I'd expect it to exit cleanly.

#### Actual behavior
Terminates with SIGSEGV and stack dump.

Same error on 2/2 shutdowns!

--- a/pkg/p2p/libp2p/libp2p.go
+++ b/pkg/p2p/libp2p/libp2p.go
@@ -53,6 +53,7 @@ type Service struct {
 	host              host.Host
 	natManager        basichost.NATManager
 	natAddrResolver   *staticAddressResolver
+	autonatDialer     host.Host
 	libp2pPeerstore   peerstore.Peerstore
 	metrics           metrics
 	networkID         uint64
@@ -219,6 +220,7 @@ func New(ctx context.Context, signer beecrypto.Signer, networkID uint64, overlay
 		host:              h,
 		natManager:        natManager,
 		natAddrResolver:   natAddrResolver,
+		autonatDialer:     dialer,
 		handshakeService:  handshakeService,
 		libp2pPeerstore:   libp2pPeerstore,
 		metrics:           newMetrics(),
@@ -703,7 +705,12 @@ func (s *Service) Close() error {
 	if err := s.libp2pPeerstore.Close(); err != nil {
 		return err
 	}
-	if err := s.natManager.Close(); err != nil {
+	if s.natManager != nil {
+		if err := s.natManager.Close(); err != nil {
+			return err
+		}
+	}
+	if err := s.autonatDialer.Close(); err != nil {
 		return err
 	}
 

